bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -331,6 +340,23 @@ public static void run(Path jbossHome, List<String> args, ModuleLoader moduleLoa
         // At this point we can configure JMX
         configureJMX(moduleClassLoader, bootableJar.log);
 
+        // Automatic loading of Security providers.
+        // Needed for logic that requires access to providers prior elytron subsystem is configured.
+        final ServiceLoader<Provider> providerServiceLoader = ServiceLoader.load(Provider.class, moduleClassLoader);
+        SecurityManager sm = System.getSecurityManager();
+        for (Provider provider : providerServiceLoader) {
```

Comment:
I believe this line can throw the RuntimeException.

The pattern in JBoss Modules installs all the providers that can be loaded, this installs all or refuses to boot.

Reply:
I don't have any argument against it either way, but the byte code ends up being pretty much the same.

Reply:
In this version the try / catch block wraps the iteration calls so will try again if one fails, in the for Provider version the try / catch block was inside the for loop so the process would exit on the first error.

The Java code could have changed since but I believe the iterator still moves to the next item even if creating it fails.